ignite/base/mixins.py

```diff
@@ -22,3 +28,262 @@ def load_state_dict(self, state_dict: Mapping) -> None:
         opts = [k in state_dict for k in self._state_dict_one_of_opt_keys]
         if len(opts) > 0 and ((not any(opts)) or (all(opts))):
             raise ValueError(f"state_dict should contain only one of '{self._state_dict_one_of_opt_keys}' keys")
+
+
+class EventsDriven:
+    """Base class for events-driven engines without state.
+
+    """
+
+    def __init__(self) -> None:
+        self._event_handlers = defaultdict(list)  # type: Dict[Any, List]
+
+        self._allowed_events = []  # type: List[EventEnum]
+        self._allowed_events_counts = {}  # type: Dict[Union[str, EventEnum], int]
+
+        self.last_event_name = None  # type: Optional[Events]
+        self.logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
```

Comment:
Could we use an ignite logger ?

Reply:
It's the same logger used in the Engine now, do we want to change it? https://github.com/pytorch/ignite/blob/master/ignite/engine/engine.py#L126

Reply:
That's the point I would like to figure out.